src/Security/Authentication/OpenIdConnect/src/OpenIdConnectConfigureOptions.cs

```diff
@@ -0,0 +1,136 @@
+        options.RefreshInterval = StringHelpers.ParseValueOrDefault(options.RefreshInterval, configSection[nameof(options.RefreshInterval)], _invariantTimeSpanParse);
+        options.RefreshOnIssuerKeyNotFound = StringHelpers.ParseValueOrDefault(options.RefreshOnIssuerKeyNotFound, configSection[nameof(options.RefreshOnIssuerKeyNotFound)], bool.Parse);
+        options.RemoteAuthenticationTimeout = StringHelpers.ParseValueOrDefault(options.RemoteAuthenticationTimeout, configSection[nameof(options.RemoteAuthenticationTimeout)], _invariantTimeSpanParse);
+        options.RemoteSignOutPath = new PathString(configSection[nameof(options.RemoteSignOutPath)] ?? options.RemoteSignOutPath);
+        options.RequireHttpsMetadata = StringHelpers.ParseValueOrDefault(options.RequireHttpsMetadata, configSection[nameof(options.RequireHttpsMetadata)], bool.Parse);
+        options.Resource = configSection[nameof(options.Resource)];
+        options.ResponseMode = configSection[nameof(options.ResponseMode)] ?? options.ResponseMode;
+        options.ResponseType = configSection[nameof(options.ResponseType)] ?? options.ResponseType;
+        options.ReturnUrlParameter = configSection[nameof(options.ReturnUrlParameter)] ?? options.ReturnUrlParameter;
+        options.SaveTokens = StringHelpers.ParseValueOrDefault(options.SaveTokens, configSection[nameof(options.SaveTokens)], bool.Parse);
+        var scopes = configSection.GetSection(nameof(options.Scope)).GetChildren().Select(scope => scope.Value).OfType<string>();
+        foreach (var scope in scopes)
+        {
+            options.Scope.Add(scope);
+        }
+        options.SignedOutCallbackPath = new PathString(configSection[nameof(options.SignedOutCallbackPath)] ?? options.SignedOutCallbackPath);
+        options.SignedOutRedirectUri = configSection[nameof(options.SignedOutRedirectUri)] ?? options.SignedOutRedirectUri;
+        options.SignInScheme = configSection[nameof(options.SignInScheme)];
+        options.SignOutScheme = configSection[nameof(options.SignOutScheme)];
+        options.SkipUnrecognizedRequests = StringHelpers.ParseValueOrDefault(options.SkipUnrecognizedRequests, configSection[nameof(options.SkipUnrecognizedRequests)], bool.Parse);
+        options.UsePkce = StringHelpers.ParseValueOrDefault(options.UsePkce, configSection[nameof(options.UsePkce)], bool.Parse);
+        options.UseTokenLifetime = StringHelpers.ParseValueOrDefault(options.UseTokenLifetime, configSection[nameof(options.UseTokenLifetime)], bool.Parse);
+    }
+
+    private static CookieBuilder GetCookieFromConfig(IConfiguration cookieConfigSection, CookieBuilder cookieBuilder)
+    {
+        if (cookieConfigSection is not null && cookieConfigSection.GetChildren().Any())
+        {
+            // Override the existing defaults when values are set instead of constructing
+            // an entirely new CookieBuilder.
+            cookieBuilder.Domain = cookieConfigSection[nameof(cookieBuilder.Domain)];
```

Comment:
Shouldn't this still fall back to `?? cookieBuilder.Domain;` if the config section doesn't provide a domain?